### PR TITLE
Keyboard Shortcuts for XKit Editor

### PR DESCRIPTION
**Pull request by Wolvan**
I added the following Keyboard Shortcuts to XKit's Extension Editor:
* Ctrl + S -> Save
* Ctrl + O -> Open
* Ctrl + E -> New Extension (can't override Ctrl + N)
* Ctrl + D -> Delete
* Ctrl + 1 -> Script Tab
* Ctrl + 2 -> CSS Tab
* Ctrl + 3 -> Icon Tab
* Ctrl + 4 -> JSON Tab

### CODE REVIEW

File: editor.js [hunk 1]
```diff
@@ -147,7 +147,16 @@ function extension_editor_finish_run() {
 	$("#xkit-editor-new").click(function() {
 
 		XKit.window.show("Create extension","<input type=\"text\" id=\"xkit-editor-filename\" placeholder=\"Filename (eg: my_extension)\"><br/>No spaces or special characters.","question","<div id=\"xkit-editor-create-extension\" class=\"xkit-button default\">OK</div><div id=\"xkit-close-message\" class=\"xkit-button\">Cancel</div>");
-
+		
+		$("#xkit-editor-filename").focus();
+		
+		$("#xkit-editor-filename").on('keydown', function(event) {
+			if(event.which == 13 || event.keyCode == 13) {
+				console.log("Pressed enter");
```
Comment (0xazure) [inline editor.js]: Probably shouldn't have a `console.log()` here.

Comment (Wolvan) [inline editor.js]: Oh shit, yeah. Sorry forgot to remove that after having to test around, fixed now

**Comment by 0xazure** [comment]
I would still recommend splitting changes to the default extension metadata into a separate PR and leave this one just for the addition of the keybindings.

From a functionality standpoint, lgtm.  Tested on Mac OSX, latest Chrome and Firefox.

Interestingly, `ctrl+1` and `ctrl+2` don't work as expected (`meta+1` and `meta+2` do) on Chrome, but they work fine on Firefox.

**Comment by 0xazure** [comment]
Only thing I can think of is adding some kind of indicator that there are keyboard shortcuts available.  There's nothing on the UI to indicate this functionality.

**Comment by Wolvan** [comment]
alright, I put them in the UI
image

File: editor.js [hunk 2]
```diff
@@ -24,21 +24,23 @@ XKit.extensions.xkit_editor = new Object({
 var script_editor, icon_editor, css_editor, object_editor;
 
 function extension_editor_run() {
-
+	
+	var mac = navigator.platform.match(/(Mac|iPhone|iPod|iPad)/i) ? "Meta" : "Ctrl";
```
Comment (0xazure) [inline editor.js]: Would probably change `mac` to `key`, or `keyText` to be more descriptive.

The `.match()` on platforms other than Mac is probably superfluous, and iOS devices don't have a `meta` key, much less `ctrl` or other modifier keys as far as I know.

**Comment by 0xazure** [comment]
lgtm :+1: @homu r+

**Pull request merged**